index.html

```diff
@@ -329,8 +331,15 @@ <h3><dfn>Quad</dfn> interface</h3>
     </pre>
 
     <p>
-      <dfn>subject</dfn> the subject, which is a <code>NamedNode</code>, <code>BlankNode</code> or
-      <code>Variable</code>.
+      <dfn>termType</dfn> contains the constant <code>"Quad"</code> or <code>undefined</code> if the
+      implementation does not support RDF*.
+    </p>
+    <p>
+      <dfn>value</dfn> contains the constant <code>undefined</code>.
```

Comment:
re @rubensworks 

> Having a value field here with a constant string or undefined feels too much like a hack to me, so I would not be in favour of that.

How about this?

```suggestion
      <dfn>value</dfn> contains a concatenation of the <code>value</code> properties of quad's subject, predicate, object and graph, separated by a single space character.
```

Reply:
Not sure what the use of this value would be. I think it would just take up memory needlessly.

Reply:
What is the use of value anyway? Would not have to be allocated up front. Make is a lazy getter?

Reply:
Indeed, it doesn't have a value, so it shouldn't exist here IMO. But the `Term` contract currently requires it. Making it optional in `Term` would also be possible, but that would lead to other problems, as consumers will never be certain that this field exist on a `Term`.

Reply:
> Indeed, it doesn't have a value

Clearly the quad **is** the value, no? As much as URI is the value of NamedNode

It was not what I asked. I meant to ask how is `Term#value` used. It has no use for equality and it is not representative of the actual value either (the way it ignores language/datatype of literals). Maybe you attribute too much importance to it right now?

> Making it optional in Term would also be possible, but that would lead to other problems

Well, yes, that would be a breaking change

Reply:
> Clearly the quad is the value, no?

Exactly. This would be a motivation for the composition approach in #163 🙂

Reply:
Fair enough

Reply:
Ah, but no

```diff
interface Term {
- attribute string value;
+ attribute (string | Quad) value;
}
```

Changing the return type of `Term#value` is even worse. This actually is a breaking change because it is a hazard for code relying on the current interface where they will expect the `value` to be a string.

Reply:
Yep, that would be a breaking change. There's no way around that in #163 AFAICS.